`CoroutinesDumpTest#testSuspendedCoroutine` failed

Failure: https://teamcity.jetbrains.com/buildConfiguration/KotlinTools_KotlinxCoroutines_Build/5116130

Stacktrace:
```
java.lang.AssertionError: Coroutine stacktrace headers are not matched:
  - Coroutine "coroutine":DeferredCoroutine{Active}@1e4a7dd4, state: SUSPENDED
  + Coroutine "coroutine":DeferredCoroutine{Active}@16ffbe93, state: RUNNING (Last suspension stacktrace, not an actual stacktrace)
 expected:<CoroutineDumpHeader(name=coroutine, className=DeferredCoroutine, state=SUSPENDED)> but was:<CoroutineDumpHeader(name=coroutine, className=DeferredCoroutine, state=RUNNING (Last suspension stacktrace, not an actual stacktrace))>
java.lang.AssertionError: Coroutine stacktrace headers are not matched:
  - Coroutine "coroutine":DeferredCoroutine{Active}@1e4a7dd4, state: SUSPENDED
  + Coroutine "coroutine":DeferredCoroutine{Active}@16ffbe93, state: RUNNING (Last suspension stacktrace, not an actual stacktrace)
 expected:<CoroutineDumpHeader(name=coroutine, className=DeferredCoroutine, state=SUSPENDED)> but was:<CoroutineDumpHeader(name=coroutine, className=DeferredCoroutine, state=RUNNING (Last suspension stacktrace, not an actual stacktrace))>
  at org.junit.Assert.fail(Assert.java:89)
  at org.junit.Assert.failNotEquals(Assert.java:835)
  at org.junit.Assert.assertEquals(Assert.java:120)
  at kotlin.test.junit.JUnitAsserter.assertEquals(JUnitSupport.kt:32)
  at kotlin.test.AssertionsKt__AssertionsKt.assertEquals(Assertions.kt:63)
  at kotlin.test.AssertionsKt.assertEquals(Unknown Source)
  at kotlinx.coroutines.debug.CoroutineDump.verify(StacktraceUtils.kt:128)
  at kotlinx.coroutines.debug.StacktraceUtilsKt.verifyDump(StacktraceUtils.kt:209)
  at kotlinx.coroutines.debug.StacktraceUtilsKt.verifyDump(StacktraceUtils.kt:42)
  at kotlinx.coroutines.debug.CoroutinesDumpTest$testSuspendedCoroutine$1.invokeSuspend(CoroutinesDumpTest.kt:28)
  at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
  at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)
  at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:256)
  at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:54)
  at kotlinx.coroutines.BuildersKt__BuildersKt.runBlockingImpl(Builders.kt:30)
  at kotlinx.coroutines.BuildersKt.runBlockingImpl(Unknown Source)
  at kotlinx.coroutines.BuildersKt__Builders_concurrentKt.runBlockingK(Builders.concurrent.kt:67)
  at kotlinx.coroutines.BuildersKt.runBlockingK(Unknown Source)
  at kotlinx.coroutines.BuildersKt__Builders_concurrentKt.runBlockingK$default(Builders.concurrent.kt:50)
  at kotlinx.coroutines.BuildersKt.runBlockingK$default(Unknown Source)
  at kotlinx.coroutines.debug.CoroutinesDumpTest.testSuspendedCoroutine(CoroutinesDumpTest.kt:21)
```

--- a/kotlinx-coroutines-debug/test/CoroutinesDumpTest.kt
+++ b/kotlinx-coroutines-debug/test/CoroutinesDumpTest.kt
@@ -209,7 +209,11 @@ class CoroutinesDumpTest : DebugTestBase() {
     private fun awaitCoroutine() = synchronized(monitor) {
         while (coroutineThread == null) (monitor as Object).wait()
         while (coroutineThread!!.state != Thread.State.TIMED_WAITING) {
-            // Wait until thread sleeps to have a consistent stacktrace
+            /* Wait until the thread we're awaiting goes to sleep.
+            Note: this does not establish a happens-before relationship with the thread entering `Thread.sleep`.
+            There still doesn't seem to be any guarantee that the stacktrace will be accurately updated.
+            <https://docs.oracle.com/javase/8/docs/api/java/lang/Thread.html#getStackTrace--> says that how accurate
+            a stacktrace is is implementation-dependent. */
         }
     }
 

